tests/expressions_window/tests.py

```diff
@@ -0,0 +1,755 @@
+import datetime
+from unittest import skipIf, skipUnless
+
+from django.core.exceptions import FieldError
+from django.db import NotSupportedError, connection
+from django.db.models import F, RowRange, ValueRange, Window
+from django.db.models.aggregates import Avg, Max, Min, Sum
+from django.db.models.expressions import ExpressionList, OrderBy, Value
+from django.db.models.functions import (
+    CumeDist, DenseRank, ExtractYear, FirstValue, Lag, LastValue, Lead,
+    NthValue, Ntile, PercentRank, Rank, RowNumber,
+)
+from django.test import TestCase, skipUnlessDBFeature
+
+from .models import WindowTestModel
+
+
+@skipUnlessDBFeature('supports_over_clause')
+class TestWindowFunction(TestCase):
```

Comment:
Do you think there might be too much test data in use here? Without a solid understanding of window functions, it becomes quite difficult to spot what the output should be, and to pick out any potential issues. You might have carefully chosen all the test data based on all the various window functions that need to be tested, so don't just change it because someone unfamiliar can't understand them.

It'd be easier to review (and to verify in the future) if the test data was smaller, and there was a one line explanation on what the results should be. For example, the `Lead` expression might say:

> Display the salary of the person in the next row, sorted by hiredate

That would be almost duplication of the documentation, but would quickly allow verification of the results without referring back to big paragraphs of text. If it's possible to keep a short explanation in one sentence (like above) I feel that would greatly help review.